jetbrains: drop bundled plugins (#470147)
NixOS/nixpkgs: pkgs/applications/editors/jetbrains/readme.md

## Tests:
- To test the build process of every IDE (as well as the process for adding plugins), build `jetbrains.plugins.tests.empty`.
- To test that plugins are correctly stored in the plugins directory, build `jetbrains.plugins.tests.stored-correctly`.

## How to use plugins:
 - Get the IDE you want and call `jetbrains.plugins.addPlugins` with a list of plugins you want to add.
 - The list has to contain contain drvs giving the directory contents of the plugin or a single `.jar` (executable).

> Nixpkgs no longer bundles plugins, however you can use third-party plugin sources, such as

Reviewer: 1. No need for "no longer"; we should document what is not what was
2. Slight wording suggestions

```suggestion
Nixpkgs does not package Jetbrains plugins, however you can use third-party sources, such as
```

Also, we should try to follow [one sentence per line](https://github.com/NixOS/nixpkgs/blob/master/doc/README.md#one-sentence-per-line) in markdown docs, such as this file. At least in new additions.

[nix-jetbrains-plugins](https://github.com/nix-community/nix-jetbrains-plugins).

> Note that some plugins may not work as-is as they are packaged in a way that isn't compatible with NixOS.

Reviewer: ```suggestion
Note that some plugins may not work without modification, if they are packaged in a way that is incompatible with NixOS.
```

You can try installing such plugins from within the IDE instead.

### Example derivations:

```nix
# "Normal" plugin
fetchzip
  {
    url = "https://plugins.jetbrains.com/files/8607/786671/NixIDEA-0.4.0.18.zip";
    hash = "sha256-JShheBoOBiWM9HubMUJvBn4H3DnWykvqPyrmetaCZiM=";
  }
  # "Single JAR file" plugin
  fetchurl
  {
    executable = true;
    url = "https://plugins.jetbrains.com/files/7425/760442/WakaTime.jar";
    hash = "sha256-DobKZKokueqq0z75d2Fo3BD8mWX9+LpGdT9C7Eu2fHc=";
  }

> ```

Reviewer: Is it better to have one codeblock per example?

Nixfmt is also treating this as passing three arguments to `fetchzip`, so another reason to use separate codeblocks.

````suggestion
#### "Normal" plugin

```nix
fetchzip {
  url = "https://plugins.jetbrains.com/files/8607/786671/NixIDEA-0.4.0.18.zip";
  hash = "sha256-JShheBoOBiWM9HubMUJvBn4H3DnWykvqPyrmetaCZiM=";
}
```

#### "Single JAR file" plugin

```nix
fetchurl {
  executable = true;
  url = "https://plugins.jetbrains.com/files/7425/760442/WakaTime.jar";
  hash = "sha256-DobKZKokueqq0z75d2Fo3BD8mWX9+LpGdT9C7Eu2fHc=";
}
```
````
